src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -131,12 +131,12 @@ public function handle(AddProductToOrderCommand $command)
 
         // update totals amount of order
         // @todo: use https://github.com/PrestaShop/decimal for prices computations
-        $order->total_products += (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
-        $order->total_products_wt += (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS);
+        $order->total_products = (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
+        $order->total_products_wt = (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS);
 
-        $order->total_paid += Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
-        $order->total_paid_tax_excl += Tools::ps_round((float) $cart->getOrderTotal(false, $totalMethod), 2);
-        $order->total_paid_tax_incl += Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
+        $order->total_paid = Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
```

Comment:
```suggestion
        $order->total_paid = Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), Cart:: Cart::ONLY_DISCOUNTS);
```

And same for other hardcoded 2 values ;)

Reply:
@matthieu-rolland The 2 value is for the `Tools::ps_round`, not the `$cart->getOrderTotal`

Reply:
ah damn, I badly read, my mistake .... in this case we need to use the proper computing precision:

`
```suggestion
        $computingPrecision = new ComputingPrecision();
        $precision = $computingPrecision->getPrecision($currency->precision);
```
Then replace 2 with $precision ...  (we're supposed to get rid of hardcoded precision everywhere)

Reply:
But I'm not blocking the PR for this since you didn't introduce this, I can do it in another PR 👍

Reply:
@matthieu-rolland I do the move now ;)